packages/protocol/contracts/signal/SignalService.sol

```diff
@@ -110,15 +109,26 @@ contract SignalService is EssentialContract, ISignalService {
         // verify that chainB's signalRoot has been sent as a signal by chainB's
         // "taiko" contract, then using chainB's signalRoot, we further check
         // the signal is sent by chainC's "bridge" contract.
-        bytes32 signalRoot = ICrossChainSync(resolve("taiko", false))
-            .getSyncedSnippet(p.height).signalRoot;
+
+        if (!isAuthorizedAs(p.crossChainSync, bytes32(block.chainid))) {
+            return false;
+        }
+
+        bytes32 signalRoot = ICrossChainSync(p.crossChainSync).getSyncedSnippet(
+            p.height
+        ).signalRoot;
+
         if (signalRoot == 0) return false;
 
         for (uint256 i; i < p.hops.length; ++i) {
             Hop memory hop = p.hops[i];
+
+            bytes32 label = authorizedAddresses[hop.signalRootRelay];
```

Comment:
what if we are doing L1 to L4, and `TaikoL2` as a predeployed proxy at the same address across multiple chains (L2, L3)? we are unable to authorize the same address for more than one chainID now?

Reply:
This is indeed a very good feedback. @davidtaikocha can we make sure the pre-deployed contracts all have unique addresses per chain, for example, if we deploy TaikoL2 on a chain with id 12345, can we use 12345 to prefix the address of the deployment of TaikoL2?

Reply:
I will update the deployment script to implement this feature.